googlemaps/places.py

```diff
@@ -0,0 +1,392 @@
+                   radius=radius, language=language, min_price=min_price,
+                   max_price=min_price, open_now=open_now, types=types,
+                   page_token=page_token)
+
+
+def places_nearby(client, location, radius=None, keyword=None, language=None,
+                  min_price=None, max_price=None, name=None, open_now=False,
+                  rank_by=None, types=None, page_token=None):
+    """
+    Performs nearby search for places.
+
+    :param location: The latitude/longitude value for which you wish to obtain the
+                     closest, human-readable address.
+    :type location: dict or list or tuple
+
+    :param radius: Distance in meters within which to bias results.
+    :type radius: int
+
+    :param keyword: A term to be matched against all content that Google has
+                    indexed for this place.
+    :type keyword: string
+
+    :param language: The language in which to return results.
+    :type langauge: string
+
+    :param min_price: Restricts results to only those places with no less than
+                      this price level. Valid values are in the range from 0
+                      (most affordable) to 4 (most expensive).
+    :type min_price: int
+
+    :param max_price: Restricts results to only those places with no greater
+                      than this price level. Valid values are in the range
+                      from 0 (most affordable) to 4 (most expensive).
+    :type max_price: int
+
+    :param name: One or more terms to be matched against the names of places.
+    :type name: string or list of strings
+
+    :param open_now: Return only those places that are open for business at
+                     the time the query is sent.
+    :type open_now: bool
+
+    :param rank_by: Specifies the order in which results are listed.
+                    Possible values are: prominence (default), distance
+    :type rank_by: string
+
+    :param types: Restricts the results to places matching at least one of the
+                  specified types.
+    :type types: string or list of strings
+
+    :param page_token: Token from a previous search that when provided will
+                       returns the next page of results for the same search.
+    :type page_token: string
+
+    :rtype: result dict with the following keys:
+            status: status code
+            results: list of places
+            html_attributions: set of attributions which must be displayed
+            next_page_token: token for retrieving the next page of results
+
+    """
+    if rank_by == "distance":
+        if not (keyword or name or types):
+          raise ValueError("either a keyword, name, or types arg is required "
+                           "when rank_by is set to distance")
+        elif radius is not None:
+          raise ValueError("radius cannot be specified when rank_by is set to "
+                           "distance")
+
```

Comment:
This actually got refactored out based off some offline feedback from @domesticmouse